lib/pbench/server/auth/__init__.py

```diff
@@ -208,32 +224,35 @@ def user_refresh_token(self, refresh_token: str) -> JSON:
         payload = {
             "client_id": self.client_id,
             "client_secret": self.client_secret_key,
-            "grant_type": ["refresh_token"],
+            "grant_type": "refresh_token",
             "refresh_token": refresh_token,
         }
         return self._post(self.TOKEN_ENDPOINT, data=payload).json()
 
     def token_introspect_online(self, token: str, token_info_uri: str) -> JSON:
         """
-        The introspection endpoint is used to retrieve the active state of a token.
+        The introspection endpoint is used to retrieve the active state of a
+        token.
         It can only be invoked by confidential clients.
-        The introspected JWT token contains the claims specified in https://tools.ietf.org/html/rfc7662
-        Note: this is not supposed to be used in production, instead rely on offline token validation
+        The introspected JWT token contains the claims specified in
+        https://tools.ietf.org/html/rfc7662
+        Note: this is not supposed to be used in production, instead rely on
+              offline token validation because of security concerns mentioned in
+              https://www.rfc-editor.org/rfc/rfc7662.html#section-4
```

Comment:
Yes, I think that is a really good point. But I think it can be handled separately with a separate Jira ticket, I will open a Jira ticket to capture this message.

Reply:
Logged it here https://issues.redhat.com/browse/PBENCH-883